blocks/rich-text/index.js

```diff
@@ -373,12 +368,22 @@ export class RichText extends Component {
 	 * Handles any case where the content of the tinyMCE instance has changed.
 	 */
 	onChange() {
-		if ( ! this.editor.isDirty() ) {
```

Comment:
The dirty checking was causing an issue when you click the "undo" button because `onChange` was being called right after the `updateContent` call. Which is a not necessary call to `onChange` (It was even causing issues).

I wonder if we should keep it to avoid these unnecessary onChange calls.

Reply:
Wouldn't `this.savedContent` checking prevent that?

Reply:
No, We're not checking it on the onChange handler when the updateContent triggers (clicking undo, splitting paragraphs)

Reply:
Okay. Looking into it. Calling save here is not great because it will call the expensive version of `getContent`... :)

Reply:
Would this do the job? https://github.com/WordPress/gutenberg/pull/4956/files#diff-766381e792acf9d4c2505f3aeddcca7eR701